annlite/executor.py

```diff
@@ -58,6 +62,11 @@ def __init__(
         self.index_traversal_paths = index_traversal_paths
         self.search_traversal_paths = search_traversal_paths
         self._valid_input_columns = ['str', 'float', 'int']
+        self._task_queue = DocumentArray()
```

Comment:
how is this a task queue?

Reply:
We found that indexing for small batches is really slow, so the task queue is used to collect data from clients and the real index operations happen in this thread(each time it will extract a relatively big batch from the queue).

Reply:
I have already added the descriptions of this pr.

Reply:
the variable name is misleading. maybe we can name it `data_buffer` to avoid ambiguity.

Reply:
Do u make sure that before search is hit, it is reindexed?

If you do so, make index endpoint async